### PR TITLE
CAP-1815 support custom sensitive annotations and labels

### PR DESCRIPTION
**Pull request by kangyili**
### What does this PR do?

Users can use `DD_ORCHESTRATOR_EXPLORER_CUSTOM_SENSITIVE_ANNOTATIONS_LABELS` to remove sensitive annotations and labels from Kubernetes resource manifests.

`DD_ORCHESTRATOR_EXPLORER_CUSTOM_SENSITIVE_ANNOTATIONS_LABELS` is a list that contains keys for sensitive annotations and labels. **_Datadog agent will scrub any annotations and labels with a matching key._**

### Example

Set the environment variable:
```
DD_ORCHESTRATOR_EXPLORER_CUSTOM_SENSITIVE_ANNOTATIONS_LABELS="sensitive-annotations sensitive-labels sensitive-key"
```

Given the following resource manifest:
```json
{
  "Annotations": {
    "my-annotation": "my-value",
    "sensitive-annotations": "sensitive-annotations-value"
    "sensitive-key":"sensitive-value"
  },
  "Labels": {
    "my-label": "my-value",
    "sensitive-labels": "sensitive-labels-value"
    "sensitive-key":"sensitive-value"
  }
}
```

The output from the agent will be:
```json
{
  "Annotations": {
    "my-annotation": "my-value",
    "sensitive-annotations": "-"
    "sensitive-key":"-"
  },
  "Labels": {
    "my-label": "my-value",
    "sensitive-labels": "-"
    "sensitive-key":"-"
  }
}
```

### CODE REVIEW

**Review by fisherevans — APPROVED** [review]

File: releasenotes/notes/support-custom-sensitive-annotations-and-labels-beb5066432315d34.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Users can use DD_ORCHESTRATOR_EXPLORER_CUSTOM_SENSITIVE_ANNOTATIONS_LABELS to remove sensitive annotations and labels.
```
Comment (neko-dd) [inline releasenotes/notes/support-custom-sensitive-annotations-and-labels-beb5066432315d34.yaml:11]:
```suggestion
    Users can use ``DD_ORCHESTRATOR_EXPLORER_CUSTOM_SENSITIVE_ANNOTATIONS_LABELS`` to remove sensitive annotations and labels.
```

Is this a parameter? A setting? A flag? How is it used? (set to `true`? something else?)

File: pkg/orchestrator/redact/pod.go
Comment (pgimalac) [inline pkg/orchestrator/redact/pod.go:1]: **💭 thought:** ‏I'm thinking you could add rules to the scrubber (`pkg/util/scrubber`) based on `orchestrator_explorer.custom_sensitive_annotations_labels`, and use it to redact your payload, but that's a pretty different way to do it so I won't block you on that 😄

**Review by pgimalac — APPROVED** [review]
LGTM

File: releasenotes/notes/support-custom-sensitive-annotations-and-labels-beb5066432315d34.yaml [hunk 2]
```diff
@@ -8,4 +8,6 @@
 ---
 enhancements:
   - |
-    Users can use DD_ORCHESTRATOR_EXPLORER_CUSTOM_SENSITIVE_ANNOTATIONS_LABELS to remove sensitive annotations and labels.
+    Users can use ``DD_ORCHESTRATOR_EXPLORER_CUSTOM_SENSITIVE_ANNOTATIONS_LABELS`` to remove sensitive annotations and labels.
+    For example: ``DD_ORCHESTRATOR_EXPLORER_CUSTOM_SENSITIVE_ANNOTATIONS_LABELS="sensitive-key-1 sensitive-key-2"``.
+    Keys should be separated by spaces. The agent will remove any annotations and labels matching these keys.
```
Comment (neko-dd) [inline releasenotes/notes/support-custom-sensitive-annotations-and-labels-beb5066432315d34.yaml:13]:
```suggestion
    Keys should be separated by spaces. The agent removes any annotations and labels matching these keys.
```

**Review by neko-dd — APPROVED** [review]
One suggestion but LGTM otherwise. The example really helps; thanks for adding it!

**Pull request merged**